src/nodes/dir.rs

```diff
@@ -363,4 +367,12 @@ impl Node for Dir {
             handle: Mutex::from(handle),
         }))
     }
+
+    fn setattr(&self, delta: &AttrDelta) -> NodeResult<fuse::FileAttr> {
+        let mut state = self.state.lock().unwrap();
+        if let Some(path) = state.underlying_path.as_ref() {
```

Comment:
Suggest using `&` here instead of `as_ref`.

Reply:
Done.